common/folderCreationTracker.go

```diff
@@ -65,8 +65,13 @@ type simpleFolderTracker struct {
 func (f *simpleFolderTracker) RecordCreation(folder string) {
 	f.mu.Lock()
 	defer f.mu.Unlock()
+	decodedFolderName, err := url.QueryUnescape(folder)
+	if err == nil {
+		f.contents[decodedFolderName] = struct{}{}
+	} else {
+		f.contents[folder] = struct{}{}
```

Comment:
This is potentially a bug. We're kind of silently ignoring error without the knowledge of surrounding code.

This is a Map lookup failure - we're using dirURL().String to lookup while we created the entry with info.Destination. Instead of specializing it for escaped characters, we should change lookup/creation to use the same string.

Reply:
Did the changes by making some adjustments in the design as we discussed. I think this should solve the bug.